clusterversion: bump minBinary to 22.2

celiala opened the pull request:
This PR bumps `binaryMinSupportedVersion` to 22.2 before cutting the 23.1 release branch, as part of the stability period process.

In previous stability periods, the work of deleting pre-22.2 gates/tests and bumping binaryMinSupportedVersion was done at the same time. For 23.1, we will break this up into smaller tasks:

* This PR will just bump `binaryMinSupportedVersion` to 22.2, skipping 22.2 mixed-version tests as needed to bump this value.
* The clean-up of pre-22.2 gates and 22.2 mixed-version tests will be done via follow-up work/issues.

Issue: #95530
Epic: CRDB-23572

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/95630)
<!-- Reviewable:end -->

celiala committed:
clusterversion: bump minBinary to 22.2

This PR bumps binaryMinSupportedVersion to 22.2 before cutting the 23.1
release branch, as part of the stability period process.

In previous stability periods, the work of deleting pre-22.2 gates/tests
and bumping binaryMinSupportedVersion was done at the same time. For 23.1,
we will break this up into smaller tasks:

- This PR will just bump binaryMinSupportedVersion to 22.2, skipping 22.2
  mixed-version tests as needed to bump this value.
- The clean-up of pre-22.2 gates and 22.2 mixed-version tests will be done
  via follow-up work/issues.

Issue: #95530
Epic: CRDB-23572

Release note: None
